pypi-dependencies.json

```diff
@@ -0,0 +1,2321 @@
+                    "sha256": "8ee45429555515e1f6b185e78100aea234072576aa43ab53aefcae078162fca9"
+                }
+            ]
+        },
+        {
+            "name": "python3-colorama",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --no-deps -I --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"colorama==0.4.6\" --no-build-isolation"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/d1/d6/3965ed04c63042e047cb6a3e6ed1a63a35087b6a609aa3a15ed8ac56c221/colorama-0.4.6-py2.py3-none-any.whl",
+                    "sha256": "4f1d9991f5acc0ca119f9d443620b77f9d6b33703e51011c16baf57afb285fc6"
+                }
+            ]
+        },
+        {
+            "name": "python3-commentjson",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --no-deps -I --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"commentjson==0.9.0\" --no-build-isolation"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/c0/76/c4aa9e408dbacee3f4de8e6c5417e5f55de7e62fb5a50300e1233a2c9cb5/commentjson-0.9.0.tar.gz",
+                    "sha256": "42f9f231d97d93aff3286a4dc0de39bfd91ae823d1d9eba9fa901fe0c7113dd4"
+                },
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/34/b8/aa7d6cf2d5efdd2fcd85cf39b33584fe12a0f7086ed451176ceb7fb510eb/lark-parser-0.7.8.tar.gz",
+                    "sha256": "26215ebb157e6fb2ee74319aa4445b9f3b7e456e26be215ce19fdaaa901c20a4"
+                }
+            ]
+        },
+        {
+            "name": "python3-crcmod",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --no-deps -I --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"crcmod==1.7\" --no-build-isolation"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/6b/b0/e595ce2a2527e169c3bcd6c33d2473c1918e0b7f6826a043ca1245dd4e5b/crcmod-1.7.tar.gz",
+                    "sha256": "dc7051a0db5f2bd48665a990d3ec1cc305a466a77358ca4492826f41f283601e"
+                }
+            ]
+        },
+        {
+            "name": "python3-cryptography",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --no-deps -I --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"cryptography==41.0.7\" --no-build-isolation"
+            ],
+            "sources": [
```

Comment:
This needs rust can a cargo manifest

Reply:
This is not entirely clear for me, I checked some other projects and it looks like rust is only needed if one builds `cryptography` from source, but here explicitly the binary `whl` is downloaded for x86_64 - which does not need a compilation so far I understand and observe.

projects compiling cryptography, thus need rust/cargo: [(1)](https://github.com/flathub/com.doycho.euterpe.gtk/blob/51664c5b7210a992faffab71e3f74a7362c07288/cryptography.json#L2) [(2)](https://github.com/flathub/org.qutebrowser.qutebrowser.Userscripts/blob/89dd583a323131974df27007a40769219b501586/python-stem/python-cryptography/python-cryptography.json)
projects pulling in the binary w/o compiling: [(1)](https://github.com/flathub/io.github.dgsasha.Remembrance/blob/master/python3-modules.json) [(2)](https://github.com/flathub/com.icanblink.blink/blob/master/com.icanblink.blink.json) [(3)](https://github.com/flathub/org.gajim.Gajim/blob/8d55c6e716bde390f0f9608c19eb9f43279aa5e9/org.gajim.Gajim.yaml#L101)

is this correct, or am I just misunderstanding something ?

Reply:
It shouldn't, it specifically prefers source packages when available https://github.com/flatpak/flatpak-builder-tools/blob/5880ad0ee58c7e8db73dae4e34fe93e458a7f205/pip/flatpak-pip-generator#L82

Reply:
You just need to change it to a source package, download it, go to the rust folder and use the cargo-generator from that repo on it to get the cargo manifest and add that here.

You linked to a bunch of manifests compiling it from source, use those as examples.

Reply:
ah, ok - will try this, thank you